tests/integration/bond_test.py

```diff
@@ -277,4 +277,49 @@ def return_bond_and_slaves_mac():
 
     return bond99_bond_and_slaves_state[Interface.MAC], \
         eth1_bond_and_slaves_state[Interface.MAC], \
-        eth2_bond_and_slaves_state[Interface.MAC]
\ No newline at end of file
+        eth2_bond_and_slaves_state[Interface.MAC]
+
+
+@pytest.fixture
+def bond99():
+    with bond_interface(name='bond99', slaves=[]) as bond_state:
+        yield bond_state
+
+
+def test_bond_and_no_slaves_have_same_mac(bond99):
+    bond99_slaves_mac, eth1_slaves_mac, eth2_slaves_mac = \
+        return_bond_and_slaves_mac()
+
+    assert not bond99_slaves_mac == eth1_slaves_mac
+    assert not eth1_slaves_mac == eth2_slaves_mac
+    assert not eth2_slaves_mac == bond99_slaves_mac
+
+
+def test_bond_and_all_removed_slaves_have_same_mac(bond99):
+    bond99[INTERFACES][0]['link-aggregation']['slaves'] = []
+    libnmstate.apply(bond99)
+    test_bond_and_no_slaves_have_same_mac(bond99)
```

Comment:
I do not understand this test. Please do not call the test function again. You can create a helper for the assert, for example `assert_all_different()` or something like this and then re-use it.